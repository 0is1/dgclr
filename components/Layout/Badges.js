// @flow
import React from 'react';
import { Badge } from 'rebass';
import { getRandomKey } from 'helpers/utils';
import type { Node } from 'react';

type Props = {
  ratings: Array<string>,
  tiny?: boolean,
};

const LayoutRatingBadges = ({ ratings, tiny }: Props) => {
  const badges: Array<?Node> = ratings.map((rating) => {
    if (rating) {
      return tiny ? (
        <Badge mx="8px" py={0} key={getRandomKey()}>
          {rating}
        </Badge>
      ) : (
        <Badge key={getRandomKey()}>{rating}</Badge>
      );
    }
    return null;
  });
  return badges;
};

LayoutRatingBadges.defaultProps = {
  tiny: false,
};

export default LayoutRatingBadges;
