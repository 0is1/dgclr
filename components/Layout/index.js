import LayoutComponent from './Layout';

export default LayoutComponent;
