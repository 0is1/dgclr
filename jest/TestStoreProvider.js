// @flow
/* eslint-env jest */
import React from 'react';
import { Provider } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import configureStore from 'redux-mock-store';

import type { Node } from 'react';

export const mockStore = configureStore();

type Props = {
  children: Node,
  storeData: {},
};

function TestStoreProvider({ children, storeData }: Props) {
  const store = mockStore(storeData);
  return <Provider store={store}>{children}</Provider>;
}

export default TestStoreProvider;
