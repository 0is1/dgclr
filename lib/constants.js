export const ADVANCED_COURSE_INFO = 'courseInfo';
export const ADVANCED_NEARBY = 'nearby';

export const RATING_OPTIONS = [
  { value: 'AAA1', label: 'AAA1' },
  { value: 'AA1', label: 'AA1' },
  { value: 'A1', label: 'A1' },
  { value: 'AA2', label: 'AA2' },
  { value: 'AA3', label: 'AA3' },
  { value: 'A2', label: 'A2' },
  { value: 'A3', label: 'A3' },
  { value: 'BB1', label: 'BB1' },
  { value: 'BB2', label: 'BB2' },
  { value: 'B1', label: 'B1' },
  { value: 'B2', label: 'B2' },
  { value: 'B3', label: 'B3' },
  { value: 'C1', label: 'C1' },
  { value: 'C2', label: 'C2' },
  { value: 'C3', label: 'C3' },
  { value: 'D1', label: 'D1' },
  { value: 'D2', label: 'D2' },
  { value: 'D3', label: 'D3' },
];

export const RATINGS_SORT_ORDER = [
  'AAA1',
  'AA1',
  'A1',
  'AA2',
  'AA3',
  'A2',
  'A3',
  'BB1',
  'BB2',
  'B1',
  'B2',
  'B3',
  'C1',
  'C2',
  'C3',
  'D1',
  'D2',
  'D3',
];

export const MAP_DEFAULT_ZOOM = 9;

export const SERVER_URL = 'https://dgclr-server.herokuapp.com/graphql';
