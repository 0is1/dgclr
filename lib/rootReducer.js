import { combineReducers } from 'redux';

import search, { initialState as searchInitialState } from 'components/SearchContainer/reducers';

export const DEFAULT_STATE = {
  search: searchInitialState,
};

export default combineReducers({
  search,
});
