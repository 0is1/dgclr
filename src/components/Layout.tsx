import React from "react";
import Head from "next/head";
import { useTranslation } from "next-i18next";
import Footer from "./Footer";
import styles from "../styles/Home.module.css";

type Props = {
  children: React.ReactNode;
};

const Layout = ({ children }: Props) => {
  const { t } = useTranslation(["common"]);
  return (
    <>
      <div className={styles.container}>
        <Head>
          <title>{t("common:title")}</title>
          <meta name="description" content={t("common:sub_title")} />
          <link rel="icon" href="/favicon.ico" />
        </Head>
        <main>{children}</main>
      </div>
      <Footer />
    </>
  );
};

export default Layout;
