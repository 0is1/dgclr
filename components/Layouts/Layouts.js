// @flow

import React from 'react';
import { connect } from 'react-redux';
import { getActiveIndex } from 'components/Tabs/selectors';
import Layout from 'components/Layout';
import { getRandomKey } from 'helpers/utils';
import type { State as ReduxState } from 'lib/types';

type Props = {
  id: string,
  layouts: [],
};
type MapStateToProps = { activeIndex: ?number };

type MapDispatchToProps = {};

type CombinedProps = Props & MapStateToProps & MapDispatchToProps;

const Layouts = ({ activeIndex, layouts }: CombinedProps) => {
  const layoutData = layouts.map<any>((layout, index: number) => (
    <Layout key={getRandomKey()} layout={layout} active={index === activeIndex} />
  ));
  return layoutData;
};

const mapStateToProps = (state: ReduxState, ownProps: Props): MapStateToProps => ({
  activeIndex: getActiveIndex(state, ownProps),
});

export default connect<CombinedProps, Props, any, any, any, Function>(mapStateToProps)(Layouts);
