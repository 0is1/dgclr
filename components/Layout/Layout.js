// @flow

import React from 'react';
import { Badge, Box, Flex } from 'rebass';
import { withTranslation } from 'lib/i18n';
import { getRandomKey } from 'helpers/utils';
import CourseStyles from 'components/Course/Course.styles';
import BaseStyles from 'components/Container/Container.styles';
import type { Layout as LayoutType } from 'lib/types';

type Props = {
  active: Boolean,
  layout: LayoutType,
  t: Function,
};

const { Strong } = CourseStyles;
const { BaseText, OL, LI } = BaseStyles;

const Layout = ({ active, layout, t }: Props) => {
  if (!active) return null;
  const { holes } = layout;
  const holeData = holes.map(hole => (
    <LI key={getRandomKey()}>
      <Flex>
        <Box width="70px">
          {!!hole.par && (
            <BaseText>
              <Strong>{t('course:par')}</Strong>
              {` ${hole.par}`}
            </BaseText>
          )}
        </Box>
        <Box width="140px">
          {!!hole.length.meter && (
            <BaseText>
              <Strong>{t('course:length')}</Strong>
              {` ${hole.length.meter}m`}
            </BaseText>
          )}
        </Box>
      </Flex>
    </LI>
  ));
  return (
    <Box width="100%" mt={2} p={2}>
      {layout.rating && <Badge mx="1rem">{layout.rating}</Badge>}
      {layout.holeCount && (
        <BaseText mt={0}>
          <Strong>{t('course:hole-count')}</Strong>
          {`${layout.holeCount}`}
        </BaseText>
      )}
      {layout.holeAverageLength && (
        <BaseText>
          <Strong>{t('course:hole-average-length')}</Strong>
          {`${layout.holeAverageLength.meter}m`}
        </BaseText>
      )}
      {layout.totalPar && (
        <BaseText>
          <Strong>{t('course:course-par')}</Strong>
          {`${layout.totalPar}`}
        </BaseText>
      )}
      {holeData && holeData.length > 0 && (
        <>
          <BaseText>
            <Strong>{t('course:course-holes')}</Strong>
          </BaseText>
          <OL>{holeData}</OL>
        </>
      )}
    </Box>
  );
};

export default withTranslation(['course'])(Layout);
