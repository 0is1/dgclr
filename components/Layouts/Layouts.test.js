/* eslint-env jest */
import React from 'react';
import { shallow, mount } from 'enzyme';
import TestStoreProvider from 'jest/TestStoreProvider';
import mockCoursesData from 'components/SearchContainer/mock/courses.mock';
import { DEFAULT_STATE } from 'lib/rootReducer';
import Layout from 'components/Layout';
import Layouts from './Layouts';

const defaultProps = {
  layouts: mockCoursesData[1].layouts,
};

function setup() {
  const storeData = {
    ...DEFAULT_STATE,
    tabs: {
      data: {
        [mockCoursesData[0]._id]: {
          id: mockCoursesData[0]._id,
          activeIndex: 0,
        },
      },
    },
  };
  const components = (
    <TestStoreProvider storeData={storeData}>
      <Layouts {...defaultProps} />
    </TestStoreProvider>
  );
  const enzymeWrapper = mount(components);
  return { enzymeWrapper };
}

describe('Layouts component', () => {
  it('Match snapshot', () => {
    const subject = shallow(
      <TestStoreProvider storeData={DEFAULT_STATE}>
        <Layouts {...defaultProps} />
      </TestStoreProvider>,
    );
    expect(subject).toMatchSnapshot();
  });
  it('Renders correctly', () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find(Layout).length).toEqual(2);
  });
});
