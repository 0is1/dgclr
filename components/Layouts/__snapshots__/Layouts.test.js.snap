// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts component Match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Layouts)
    layouts={
      Array [
        Object {
          "_id": "5b86d8ea88ad697a0c00b99e",
          "holeCount": 18,
          "holes": Array [
            Object {
              "length": Object {
                "foot": 432,
                "meter": 132,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 183,
                "meter": 56,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 590,
                "meter": 180,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 242,
                "meter": 74,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 531,
                "meter": 162,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 275,
                "meter": 84,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 308,
                "meter": 94,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 295,
                "meter": 90,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 252,
                "meter": 77,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 393,
                "meter": 120,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 698,
                "meter": 213,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 311,
                "meter": 95,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 308,
                "meter": 94,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 403,
                "meter": 123,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 334,
                "meter": 102,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 328,
                "meter": 100,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 406,
                "meter": 124,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 324,
                "meter": 99,
              },
              "par": 3,
            },
          ],
          "name": "Ford DiscGolfPark, Punainen",
          "rating": "AA1",
          "totalPar": 57,
        },
        Object {
          "_id": "5b86d8ea88ad697a0c00b99d",
          "holeCount": 18,
          "holes": Array [
            Object {
              "length": Object {
                "foot": 724,
                "meter": 221,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 229,
                "meter": 70,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 590,
                "meter": 180,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 242,
                "meter": 74,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 531,
                "meter": 162,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 275,
                "meter": 84,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 308,
                "meter": 94,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 295,
                "meter": 90,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 255,
                "meter": 78,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 393,
                "meter": 120,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 698,
                "meter": 213,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 459,
                "meter": 140,
              },
              "par": 4,
            },
            Object {
              "length": Object {
                "foot": 308,
                "meter": 94,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 442,
                "meter": 135,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 334,
                "meter": 102,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 328,
                "meter": 100,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 419,
                "meter": 128,
              },
              "par": 3,
            },
            Object {
              "length": Object {
                "foot": 324,
                "meter": 99,
              },
              "par": 3,
            },
          ],
          "name": "Ford DiscGolfPark, Sininen",
          "rating": "AA1",
          "totalPar": 59,
        },
      ]
    }
  />
</Provider>
`;
